.github/workflows/check.yaml

```diff
@@ -19,27 +23,20 @@ jobs:
         python-version: ["3.8", "3.10"]
     with:
       python-version: ${{ matrix.python-version }}
-      tox-version: "<4"
 
   func:
     uses: canonical/bootstack-actions/.github/workflows/func.yaml@v3
     needs: lint-unit
     strategy:
       fail-fast: false
+      matrix:
+        command: ["TEST_JUJU3=1 make functional"]  # TEST_JUJU3 needed due https://github.com/openstack-charmers/zaza/blob/b22c2eed4c322f1dfc14ffb2d31e0dd18c911a40/setup.py#L47 to support Juju3+
+        juju-channel: ["3.4/stable"]
     with:
-      command: "make functional"
-      juju-channel: "3.1/stable"
+      command: ${{ matrix.command }}
+      juju-channel: ${{ matrix.juju-channel }}
       nested-containers: false
       provider: "lxd"
       python-version: "3.10"
       timeout-minutes: 120
-      tox-version: "<4"
-      runs-on: "['self-hosted', 'runner-k8s-svc-check']"
-      action-operator: false
-      external-controller: true
-      juju-controller: soleng-ci-ctrl
-      zaza-yaml: "LS0tCm1vZGVsX3NldHRpbmdzOgogIGltYWdlLXN0cmVhbTogcmVsZWFzZWQKcmVnaW9uOiBwcm9kc3RhY2s2CmNsb3VkOiBidWlsZGVyLWNsb3VkCmNyZWRlbnRpYWw6IGJ1aWxkZXItY2xvdWQtY3JlZAo="
-    secrets:
-      juju-controllers-yaml: ${{ secrets.JUJU_CONTROLLERS_YAML }}
-      juju-accounts-yaml: ${{ secrets.JUJU_ACCOUNTS_YAML }}
-      openstack-auth-env: ${{ secrets.OPENSTACK_AUTH_ENV }}
+      runs-on: "['self-hosted', 'linux', 'x64', 'large']]"
```

Comment:
**praise:** 

I am surprise it can run on lxd cloud. Nice one here.

Reply:
It's because `lxd-profile.yaml` is provided in kubernetes-worker and kubernetes-control-plane, so they can be deployed on top of LXD. Official runner with 4cpu was not enough, our own with 8 is working fine.